onnxruntime/test/contrib_ops/col2im_test.cc

```diff
@@ -0,0 +1,201 @@
+                              142., 0., 146., 0., 150.,
+                              0., 0., 0., 0., 0.};
+  test.AddInput<float>("input", {2, 4, 16},  input);
+  test.AddInput<int64_t>("image_shape", {2},  std::vector<int64_t>{4, 5});
+  test.AddInput<int64_t>("block_shape", {2},  std::vector<int64_t>{1, 2});
+
+  test.AddOutput<float>("output", {2, 2, 4, 5}, output);
+  test.Run();
+}
+
+TEST(Col2ImContribOpTest, with3channels4dNCHW) {
+  OpTester test("Col2Im", 1, kMSDomain);
+
+  test.AddAttribute("strides", std::vector<int64_t>{1, 1});
+  test.AddAttribute("dilations", std::vector<int64_t>{1, 1});
+  test.AddAttribute("pads", std::vector<int64_t>{0, 0, 0, 0});
+
+  std::vector<float> input(75);
+  std::vector<float> output(75);
+  std::iota(output.begin(), output.end(), float_iota(1., 1.));
+  input = _transpose_serialized_vector(output, 1, 3, 5, 5);
+  test.AddInput<float>("input", {1, 15, 5},  input);
+  test.AddInput<int64_t>("image_shape", {2},  std::vector<int64_t>{5, 5});
+  test.AddInput<int64_t>("block_shape", {2},  std::vector<int64_t>{1, 5});
+
+  test.AddOutput<float>("output", {1, 3, 5, 5}, output);
+  test.Run();
+}
+
+TEST(Col2ImContribOpTest, with2Images3channels4dNCHW) {
+  OpTester test("Col2Im", 1, kMSDomain);
+
+  test.AddAttribute("strides", std::vector<int64_t>{1, 1});
+  test.AddAttribute("dilations", std::vector<int64_t>{1, 1});
+  test.AddAttribute("pads", std::vector<int64_t>{0, 0, 0, 0});
+
+  std::vector<float> input(150);
+  std::vector<float> output(150);
+  std::iota(output.begin(), output.end(), float_iota(1., 1.));
+  input = _transpose_serialized_vector(output, 2, 3, 5, 5);
+  test.AddInput<float>("input", {2, 15, 5},  input);
+  test.AddInput<int64_t>("image_shape", {2},  std::vector<int64_t>{5, 5});
+  test.AddInput<int64_t>("block_shape", {2},  std::vector<int64_t>{1, 5});
+
+  test.AddOutput<float>("output", {2, 3, 5, 5}, output);
+  test.Run();
+}
+
+TEST(Col2ImContribOpTest, simple5dNCHWD) {
+  OpTester test("Col2Im", 1, kMSDomain);
+
+  test.AddAttribute("strides", std::vector<int64_t>{1, 1, 1});
+  test.AddAttribute("dilations", std::vector<int64_t>{1, 1, 1});
+  test.AddAttribute("pads", std::vector<int64_t>{0, 0, 0, 0, 0, 0});
+
+  std::vector<float> input(25);
+  std::vector<float> output(25);
+  std::iota(output.begin(), output.end(), float_iota(1., 1.));
+  input = _transpose_serialized_vector(output, 1, 1, 5, 5);
+  test.AddInput<float>("input", {1, 5, 5},  input);
+  test.AddInput<int64_t>("image_shape", {3},  std::vector<int64_t>{1, 5, 5});
+  test.AddInput<int64_t>("block_shape", {3},  std::vector<int64_t>{1, 1, 5});
+  test.AddOutput<float>("output", {1, 1, 1, 5, 5}, output);
+  test.Run();
+}
+
+TEST(Im2ColContribOpTest, simple) {
+  std::vector<float> input(24);
+  std::vector<float> expected_output(24);
+  std::iota(input.begin(), input.end(), float_iota(1., 1.));
+  expected_output = {1, 5, 9, 2, 6, 10, 3, 7, 11, 4, 8, 12, 13, 17, 21, 14, 18, 22, 15, 19, 23, 16, 20, 24};
```

Comment:
Is this compared against?

Reply:
during-debug-test code, not needed for the pr